### PR TITLE
Logic to rebroadcast transaction for greater chance of success

### PR DESCRIPTION
**Pull request by westleysys**
## Ticket
*N/A

## Changes
*While waiting for a transaction to confirm, rebroadcast it on regular intervals to improve chance of successfully relaying to the network.

### CODE REVIEW

File: src/renderer/constants.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ const intervals = {
   NETWORK_STATUS: 10 * 1000,
   POLLING: 15000,
   WALLET_VERSION_CHECK: 10 * 60 * 1000,
+  REBROADCAST_TRANSACTIONS: 30 * 1000,
```
Comment (crishellco) [inline src/renderer/constants.js]: Alphabetize

File: src/renderer/constants.js [hunk 2]
```diff
@@ -44,6 +45,7 @@ const timeouts = {
   RENDER_CHART: 100,
   TRANSACTION: 5 * 60 * 1000,
   TRANSACTION_WITH_HARDWARE: 8 * 60 * 1000,
+  MONITOR_TRANSACTIONS: 10 * 60 * 1000,
```
Comment (crishellco) [inline src/renderer/constants.js]: Alphabetize

File: src/renderer/services/neo.js [hunk 3]
```diff
@@ -655,16 +655,24 @@ export default {
...
               return reject('Failed to create transaction.');
             }
-            alerts.success(`Transaction Hash: ${res.tx.hash} Sent, waiting for confirmation.`);
+
+            if (res.response.result !== true) {
+              this.$store.commit('setSendInProgress', false);
+              return reject('Transaction rejected by NEO network.');
+            }
+
+            alerts.success(`Transaction Hash: ${res.tx.hash} Successfully Sent, waiting for confirmation.`);
```
Comment (crishellco) [inline src/renderer/services/neo.js]: Eventually we need to move all of these alerts out of these services. They are moving to a shared npm package to be used with the mobile/web app.

File: src/renderer/services/neo.js [hunk 4]
```diff
@@ -783,19 +793,35 @@ export default {
...
+              tx.lastBroadcasted = moment().utc();
+              api.sendTx({
+                tx,
+                url: network.getSelectedNetwork().rpc,
+              });
+              return;
+            }
+
+            const txInHistory = _.find(store.state.recentTransactions, (o) => {
+              return o.hash === tx.hash;
```
Comment (crishellco) [inline src/renderer/services/neo.js]: `const txInHistory = _.find(store.state.recentTransactions, {hash: tx.hash});`

https://lodash.com/docs/4.17.10#find

File: src/renderer/services/neo.js [hunk 5]
```diff
@@ -857,6 +884,7 @@ export default {
         alerts.networkException(e);
         lastClaimSent = null;
         store.commit('setGasClaim', gasClaim);
+        store.commit('setShowClaimGasModal', false);
```
Comment (crishellco) [inline src/renderer/services/neo.js]: Eventually we need to move all of these store mutations out of these services. They are moving to a shared npm package to be used with the mobile/web app.

File: src/renderer/services/neo.js [hunk 6]
```diff
@@ -783,19 +793,33 @@ export default {
...
       try {
         setTimeout(() => {
+          const startedMonitoring = moment().utc();
           const interval = setInterval(() => {
-            const tx = _.find(store.state.recentTransactions, (o) => {
-              return o.hash === hash;
-            });
+            if (moment().utc().diff(startedMonitoring, 'milliseconds') > timeouts.MONITOR_TRANSACTIONS) {
+              clearInterval(interval);
+              reject('Timed out waiting for transaction to be returned third party block explorer');
```
Comment (jsolman) [inline src/renderer/services/neo.js]: typo: returned third -> returned from third

**Pull request merged**